### PR TITLE
Api for Application deployment with EzAppConfig using CLI

### PR DESCRIPTION
**Pull request by oshep**

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# API
+
+This scripts are designed to simplify the deployment and configuration of an application using EzAppConfig and Helm charts.
+API folder contains the scripts and configuration files related to the EzAppConfig cli API.
+
+## Prerequisites
```
Comment (prakashmirji) [inline api/README.md:19]: mention about kubeconfig
linux or mac machine ( windows not tested )

File: api/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Before running the scripts in this folder, ensure that the following tools are installed:
+
+- [helm](https://github.com/helm/helm/releases/tag/v3.14.2)
```
Comment (prakashmirji) [inline api/README.md]: mention about the version used/tested

File: api/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+### Delete the chart tarball from Chartmuseum:
+
+```console
+delete_chart_bundle [<chart_version>]
+```
+ The optional <chart_version> parameter is used to specify a specific chart version to delete. If not provided, the current version from `${CHART_PATH}/Chart.yaml` will be used.
+
+
+## Configuration
```
Comment (prakashmirji) [inline api/README.md:82]: should go up

File: api/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        install
+        upgrade
+        delete
+        build_and_upload_chart_bundle
+                Uploads the current chart version to chartmuseum
+        delete_chart_bundle [chart_version]
+                Deletes the chart version from chartmuseum, if chart_version is not provided, current version is passed
+```
+
+Before you begin installing the application, please read the [configuration](#configuration) section.
```
Comment (prakashmirji) [inline api/README.md]: general idea or steps:

How to install app

How to upgrade app

How to delete app

Optional/Troubleshooting for chartmuseum

how to upload chart

how to delete chart

File: api/tooling/common.sh [hunk 5]
```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
```
Comment (prakashmirji) [inline api/tooling/common.sh:1]: mention, don't change this file

File: api/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
+# API
```
Comment (prakashmirji) [inline api/README.md:1]: we need another section which talks about summary
- things to do like cloning
- configuration changes
- running ..etc
- we also talk about EzAppConfig CR

**Review by akravacyber — APPROVED** [review]

**Pull request merged**